network/basics/uri.py

```diff
@@ -40,6 +40,12 @@
         directory, the basename of the file on the remote server will be used.
     required: false
     default: null
+  source:
+    description:
+      - path to binary file to be uploaded to the endpoint
```

Comment:
This doesn't seem to need to document itself as being only for binary files.  It looks like it will work with any file.